File: specification/specs2/cluster/cluster_allocation_explain/ClusterAllocationExplainRequest.ts
@rest_spec_name("cluster.allocation_explain")
class ClusterAllocationExplainRequest extends RequestBase {
	index: IndexName;
	shard: integer;
	primary: boolean;
	@request_parameter()
	include_yes_decisions: boolean;
	@request_parameter()
	include_disk_info: boolean;
}

File: output/csharp/Cluster/ClusterAllocationExplain/ClusterAllocationExplainRequest.cs
using Nest.CommonAbstractions;
using Nest.Internal;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.Cluster {

	public class ClusterAllocationExplainRequest : RequestBase {
		
		[DataMember(Name="include_disk_info")]
		public bool IncludeDiskInfo { get; set; }


		[DataMember(Name="include_yes_decisions")]
		public bool IncludeYesDecisions { get; set; }


		[DataMember(Name="index")]
		public IndexName Index { get; set; }


		[DataMember(Name="primary")]
		public bool Primary { get; set; }


		[DataMember(Name="shard")]
		public int Shard { get; set; }

	}
}
